### PR TITLE
[stable/fluentd-elasticsearch] Add DaemonSet updateStartegy settings to values.yml

### LINKED ISSUE #7978: [stable/fluentd-elasticsearch] Add updateStrategy to DaemonSet
Feature request

Version of Helm and Kubernetes:
Helm: 2.10.0
K8S: 1.10.6

Which chart:
stable/fluentd-elasticsearch

What happened:

By default DaemonSetsets created by this chart have updateStrategy: onDelete

What you expected to happen:

I'd like to configure DaemonSet via values.yaml

### PR DESCRIPTION
**Pull request by francois-travais**
Signed-off-by: François Travais <francois.travais@gmail.com>

#### What this PR does / why we need it:

Make DaemonSet updateStartegy configurable from values.yml

#### Which issue this PR fixes
  - fixes #7978 

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/fluentd-elasticsearch/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: fluentd-elasticsearch
-version: 1.0.3
+version: 1.0.4
```
Comment (desaintmartin) [inline stable/fluentd-elasticsearch/Chart.yaml]: Since it is a new feature, it may be nice to bump the minor version instead of patch.

**Comment by maver1ck** [comment]
@francois-travais Could you please fix conflict ?

**Comment by francois-travais** [comment]
Done.

**Review by paulczar — CHANGES_REQUESTED** [review]
Looks like you need to fix the length of the persistent volume name ...  see errors in test

```
I1115 17:24:13.935] Error: release fluentd-elasticsearch-f378qxbczk failed: DaemonSet.apps "fluentd-elasticsearch-f378qxbczk-fluentd-elasticsearch" is invalid: [spec.template.spec.volumes[3].name: Invalid value: "config-volume-fluentd-elasticsearch-f378qxbczk-fluentd-elasticsearch": must be no more than 63 characters, spec.template.spec.containers[0].volumeMounts[3].name: Not found: "config-volume-fluentd-elasticsearch-f378qxbczk-fluentd-elasticsearch"]
```

**Comment by francois-travais** [comment]
That's a bit odd since it has nothing to do with my changes

**Comment by francois-travais** [comment]
It's now fixed @paulczar

**Comment by maver1ck** [comment]
@francois-travais 
1) Please fix conflict
2) Should we have default value for update strategy ? So RollingUpdate instead of `{}`

File: stable/fluentd-elasticsearch/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: fluentd-elasticsearch
-version: 1.3.0
+version: 1.4.0
```
Comment (ScottBrenner) [inline stable/fluentd-elasticsearch/Chart.yaml]: Is there a `1.4.0` version of this chart? Latest I'm seeing is `1.3.0`.

**Comment by francois-travais** [comment]
@desaintmartin suggested to bump the minor version a while ago
https://github.com/helm/charts/pull/8302#discussion_r223765850, which I
did, so indeed there isn't a version 1.4.0 right now, but that's intended.

**Comment by maver1ck** [comment]
/lgtm
But I still don't have rights.

**Comment by maver1ck** [comment]
@paulczar @viglesiasce 
Could you please merge this ?

**Review by paulczar — APPROVED** [review]

**Comment by desaintmartin** [comment]
chart.yaml: appVersion did change but version did not.

**Comment by paulczar** [comment]
@desaintmartin yeah i think it did, some weird result of resolving conflicts.

**Pull request merged**